extension/docs/IntegrationGuide.md

```diff
@@ -0,0 +1,112 @@
+    - [Recalculate cart](#recalculate-cart)
+    - [Validate payment](#validate-payment)
+    - [Validate payment transaction](#validate-payment-transaction)
+    - [Mapping from Adyen result codes to CTP transaction state](#mapping-from-adyen-result-codes-to-ctp-transaction-state)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Glossary
+In this process, there are 3 parties involved:
+
+**Frontend** - the browser part of the shop. This is what the shopper interacts with.  
+**Backend** - the shop server which supplies front end with data.  
+**Adyen-integration** - hosted service (this repository) that interacts over [API extensions](https://docs.commercetools.com/http-api-projects-api-extensions).  
+**Shopper** - a person that's using the shop
+
+## Requirements for CTP project:
+All the requirements below should be automatically created by the Adyen-integration. It will not create if there
+are already resources with same key in the CTP project. In this case, you have to synchronize by yourself.
+1. [API Extension subscription to Adyen-integration endpoints](../resources/api-extensions.json)
+1. [Custom types for payments](../resources/payment-custom-types.json)
+1. [Custom types for interface interactions](../resources/payment-interface-interaction-types.json)
+
+## Required parameters
+In order to make the extension module working, following parameters have to be provided to the Adyen-integration.
+- `CTP_PROJECT_KEY`, `CTP_CLIENT_ID`, `CTP_CLIENT_SECRET` - CTP credentials of the shop project. Go to `https://mc.commercetools.com/${your-ctp-project}/settings/developer/api-clients`. This module needs to CRUD multiple CTP resources, thus recommended scope is manage_project.
+- `API_EXTENSION_BASE_URL` - URL of the Adyen-integration. This URL will be called by CTP Extension endpoint.
+- `ADYEN_API_KEY` - Go to [Account/Users](https://ca-test.adyen.com/ca/ca/config/users.shtml) - Select a user with `Web Service` User type - Generate New API Key (notice: in case you get `403 Forbidden` error from Adyen, try to regenerate the key)   
+- `ADYEN_MERCHANT_ACCOUNT` - Go to [Account/Merchant accounts](https://ca-test.adyen.com/ca/ca/accounts/show.shtml?accountTypeCode=MerchantAccount)
+
+# Checkout steps
+1. On each checkout step [validate cart state](#validate-cart-state)
+1. Before starting payment process make sure there is no valid payments already:
+    * [Recalculate cart](#recalculate-cart)
+    * [Validate payment](#validate-payment)
+    * [Validate payment transaction](#validate-payment-transaction)
+
+If all above validations are passed then order can be created right away and order confirmation page shown.
+Otherwise shopper might continue with further payment steps.
+
+1. **Get available payment methods**  
+    1. In order to [get which payment methods are available to the current shopper](https://docs.adyen.com/developers/checkout/api-integration#step1getavailablepaymentmethods),
```

Comment:
I would like to read more simple explanation with an action like: 
- **create/update** a CTP payment resource with following [paymentMethodInfo](https://docs.commercetools.com/http-api-projects-payments#paymentmethodinfo)